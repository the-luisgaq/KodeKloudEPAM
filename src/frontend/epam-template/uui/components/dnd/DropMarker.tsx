import * as React from 'react';
import cx from 'classnames';
import { DndActorRenderParams, IHasCX } from '@epam/uui-core';
import css from './DropMarker.module.scss';

interface DropMarkerProps extends DndActorRenderParams, IHasCX {
    /**
     * Pass 'true' to enable Blocker.
     */
    enableBlocker?: boolean;
}

export function DropMarker(props: DropMarkerProps) {
    return props.isDndInProgress
        ? (
            <>
                { props.enableBlocker && <div className={ css.blocker } /> }
                <div className={ cx([
                    css.root,
                    css.marker,
                    css[props.position],
                    props?.cx,
                ]) }
                />
            </>
        )
        : null;
}
