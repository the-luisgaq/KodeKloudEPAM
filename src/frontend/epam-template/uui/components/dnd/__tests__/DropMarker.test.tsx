import React from 'react';
import { render } from '@epam/uui-test-utils';
import { DropMarker } from '../DropMarker';

describe('DropMarker', () => {
    it('should be rendered correctly', () => {
        const mockEventHandlers: any = {};
        mockEventHandlers.onMouseDown = jest.fn();
        mockEventHandlers.onMouseEnter = jest.fn();
        mockEventHandlers.onMouseMove = jest.fn();
        mockEventHandlers.onMouseLeave = jest.fn();
        mockEventHandlers.onMouseUp = jest.fn();

        const { asFragment } = render(
            <DropMarker
                classNames={ ['test-class'] }
                eventHandlers={ mockEventHandlers }
                isDndInProgress
                isDraggable
                isDraggedOut
                isDraggedOver
                isDragGhost
                isDropAccepted
                position="top"
            />,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
