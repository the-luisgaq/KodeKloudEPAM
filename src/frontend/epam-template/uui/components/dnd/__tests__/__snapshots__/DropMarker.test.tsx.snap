// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropMarker should be rendered correctly 1`] = `
<DocumentFragment>
  <div
    class="root marker top"
  />
</DocumentFragment>
`;
